packages/dd-trace/src/llmobs/tagger.js

```diff
@@ -188,97 +182,92 @@ class LLMObsTagger {
+        }
 
-          if (message == null || typeof message !== 'object') {
-            logger.warn('Messages must be a string, object, or list of objects')
-            return undefined
-          }
+        if (message == null || typeof message !== 'object') {
+          logger.warn('Messages must be a string, object, or list of objects')
+          return undefined
+        }
+
+        const { content = '', role } = message
+        let toolCalls = message.toolCalls
+        const messageObj = { content }
 
-          const { content = '', role } = message
-          let toolCalls = message.toolCalls
-          const messageObj = { content }
+        if (typeof content !== 'string') {
+          logger.warn('Message content must be a string.')
+          return undefined
+        }
 
-          if (typeof content !== 'string') {
-            logger.warn('Message content must be a string.')
+        if (role) {
+          if (typeof role !== 'string') {
+            logger.warn('Message role must be a string.')
             return undefined
           }
+          messageObj.role = role
+        }
 
-          if (role) {
-            if (typeof role !== 'string') {
-              logger.warn('Message role must be a string.')
-              return undefined
-            }
-            messageObj.role = role
+        if (toolCalls) {
+          if (!Array.isArray(toolCalls)) {
+            toolCalls = [toolCalls]
           }
 
-          if (toolCalls) {
-            if (!Array.isArray(toolCalls)) {
-              toolCalls = [toolCalls]
+          const filteredToolCalls = toolCalls.map(toolCall => {
+            if (typeof toolCall !== 'object') {
+              logger.warn('Tool call must be an object.')
+              return undefined
             }
 
-            const filteredToolCalls = toolCalls.map(toolCall => {
-              if (typeof toolCall !== 'object') {
-                logger.warn('Tool call must be an object.')
-                return undefined
-              }
+            const { name, arguments: args, toolId, type } = toolCall
+            const toolCallObj = {}
 
-              const { name, arguments: args, toolId, type } = toolCall
-              const toolCallObj = {}
-
-              if (name) {
-                if (typeof name !== 'string') {
-                  logger.warn('Tool name must be a string.')
-                  return undefined
-                }
-                toolCallObj.name = name
+            if (name) {
+              if (typeof name !== 'string') {
+                logger.warn('Tool name must be a string.')
+                return undefined
               }
+              toolCallObj.name = name
+            }
 
-              if (args) {
-                if (typeof args !== 'object') {
-                  logger.warn('Tool arguments must be an object.')
-                  return undefined
-                }
-                toolCallObj.arguments = args
+            if (args) {
```

Comment:
Instead of repeating this condition several times it could be extracted to a function.

Reply:
addressed in 80168f80aa4218c51a3bb7789ae49888d69f8627